FreeRTOS/FreeRTOS
# Return error if invalid input detected in transport layer (Send/Recv)

Return error if invalid input detected in transport layer

Description
-----------
Check the input at *FreeRTOS_send/recv function and return negative value if input is invalid.

Test Steps
-----------
1. Open Windows simulator to test different transport layers.
2. Send/Receive with valid/invalid inputs
3. Assert happened when testing invalid inputs.
using_mbedtls_pcks11: corePKCS11_MQTT_Mutual_Auth_Windows_Simulator
using_mbedtls: FreeRTOS_Cellular_Interface_Windows_Simulator
using_plaintext: coreMQTT_Windows_Simulator/MQTT_Plain_Text
using_wolfSSL: coreMQTT_Windows_Simulator/MQTT_Mutual_Auth_wolfSSL

Related Issue
-----------
Refer to [API reference](https://www.freertos.org/network-interface.html)

Openssl_Recv should return negative value when byteToRecv is 0.
Openssl_Send should return negative value when byteToSend is 0.

TransportRecv_t:

If no data is available on the network to read and no error has occurred, zero MUST be the return value. A zero return value SHOULD represent that the read operation can be retried by calling the API function. Zero MUST NOT be returned if a network disconnection has occurred.

TransportSend_t:

If no data is transmitted over the network due to a full TX buffer and no network error has occurred, this MUST return zero as the return value. A zero return value SHOULD represent that the send operation can be retried by calling the API function. Zero MUST NOT be returned if a network disconnection has occurred.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
